### ISSUE TITLE
IBrowserFetcher is missing some methods on BrowserFetcher

### ISSUE THREAD

**Issue by BaileyMillerSSI** [hardkoded/puppeteer-sharp#2359]
## Looking for some help? Get priority support by [sponsoring the project](https://github.com/sponsors/hardkoded)/

**Before you file a bug, have you:**

* Tried upgrading to newest version of PuppeteerSharp, to see if your issue has already been resolved and released?
    * That is the problem
* Checked existing open *and* closed [issues](https://github.com/kblok/puppeteer-sharp/issues?utf8=%E2%9C%93&q=is%3Aissue), to see if the issue has already been reported?
    * Don't see any mention of this.
* Tried reproducing your problem in a new isolated project?
    * N/A
* Considered if this is a general question and not a bug?. For general questions please use [StackOverflow](https://stackoverflow.com/search?q=puppeteer-sharp).

### Description
TLDR: Prior to version 11 I could pass interfaces around which allows for easier unit testing. After version 11 GetExecutablePath is no longer available on IBrowserFetcher, which breaks passing interfaces.

Previously I was running on version 9.1 which had GetExecutablePath on the interface IBrowserFetcher. After switching to version >=11 I noticed that this has been removed from the interface but is available on the base class BrowserFetcher. This might not cause any direct problems but it makes unit testing this library difficult. Before when it was on the interface it was easy to mock instances.

### Complete minimal example reproducing the issue

Complete means the code snippet can be copied into a unit test method in a fresh C# project and run.
Minimal means it is stripped from code not related to reproducing the issue.

E.g.

```csharp
async Task Main()
{
	using (var browserFetcher = CreateFeather())
	{
		await browserFetcher.DownloadAsync(BrowserFetcher.DefaultChromiumRevision);

		using (var browser = await CreateBrowser(browserFetcher))
		using (var page = await browser.NewPageAsync())
		{
			await page.GoToAsync("https://www.google.com", WaitUntilNavigation.Networkidle0);
		}
	}
}

private static IBrowserFetcher CreateFeather() =>
	new BrowserFetcher();
	
private static Task<IBrowser> CreateBrowser(IBrowserFetcher fetcher)
{
	var launchOptions = new LaunchOptions()
	{
		Headless = true,
		ExecutablePath = fetcher.GetExecutablePath(BrowserFetcher.DefaultChromiumRevision)
	};

	return Puppeteer.LaunchAsync(launchOptions);
}
```

### Expected behavior:

IBrowserFetcher will have the same methods that BrowserFetcher has in respect to public methods.

### Actual behavior:

GetExecutablePath isn't available on the IBrowserFetcher interface.

### Versions

* Which version of PuppeteerSharp are you using?
    * Coming from <11, issues started after version 11 and continues into the latest version 12
* Which .NET runtime and version are you targeting? E.g. .NET framework 4.6.1 or .NET Core 2.0.
    * Net6

**Comment by saptarshidas1809** [comment]
Hi, I went through `BrowserFetcher.cs`, the method `GetExecutablePath` and `GetCurrentPlatform` are marked as internal. Is there a specific reason why this is done? 
1) It could be that `IBrowserFetcher` may be implemented by other classes as well, where these methods are not to be accessed (for whatever reason). In that case, we can version out `IBrowserFetcher` and have it inherit this which in turn could be implemented by `BrowserFetcher`. 
2) If point 1 is not applicable, then we can just add these methods into the interface.

**Comment by kblok** [comment]
@saptarshidas1809 `GetExecutablePath` was replaced with the property `CacheDir`. `GetExecutablePath` is now used to set a default value to `CacheDir` if `options.Path` was not set.

**Comment by kblok** [comment]
@BaileyMillerSSI would you mind taking a look at https://github.com/hardkoded/puppeteer-sharp/pull/2363 ?

**Comment by BaileyMillerSSI** [comment]
@kblok 
I noticed that DownloadsFolder is also gone from IBrowserFetcher. Is it safe to assume that this is now CacheDir?

**Comment by kblok** [comment]
@BaileyMillerSSI, The introduction of new browser builds changed how directories are being created, matching puppeteer.
The `CacheDir` is the root directory. But consider it won't have the full path. E.g.

![image](https://github.com/hardkoded/puppeteer-sharp/assets/2198466/08b84495-a38c-4d32-bd2b-596526faada9)


I think that now the best place to find that logic is in the [Cache](https://github.com/hardkoded/puppeteer-sharp/blob/master/lib/PuppeteerSharp/BrowserData/Cache.cs#L18) class. We can make that class public if you need to.

**Comment by BaileyMillerSSI** [comment]
@kblok 
I "think" the CacheDir does what I was looking for. Essentially I have this service that when running locally it can store the download wherever but when running on the server it needs a set path to re-use the same download across app instances.

My updated test:

```
[Theory]
[InlineData("Local")]
[InlineData("Development")]
public void ItWillCreateFetcherCorrectly(string environmentName)
{
    var hostEnvironmentMock = new Mock<IHostEnvironment>();
    hostEnvironmentMock
        .SetupGet(service => service.EnvironmentName)
        .Returns(environmentName);

    var puppeteerService = new PuppeteerService(hostEnvironmentMock.Object);

    using (var browserFetcher = puppeteerService.CreateFetcher())
    {
        var isLocal = hostEnvironmentMock.Object.IsLocal();

        if (isLocal)
        {
            browserFetcher
                .CacheDir
                .Should()
                .NotBe(PuppeteerService.ServerDownloadPath);
        }
        else
        {
            browserFetcher
                .CacheDir
                .Should()
                .Be(PuppeteerService.ServerDownloadPath);
        }
    }
}
```

My service that re-uses the Chrome download on the server with a set path:

```
public class PuppeteerService : IPuppeteerService
{
    public const string ServerDownloadPath = "/srv/app_storage/appX/";

    private readonly IHostEnvironment _environment;

    public PuppeteerService(IHostEnvironment environment)
    {
        _environment = environment;
    }

    public IBrowserFetcher CreateFetcher() =>
        new BrowserFetcher(new BrowserFetcherOptions
        {
            Path = !_environment.IsLocal() ? ServerDownloadPath : null
        });

    public async Task<IBrowser> CreateBrowser(IBrowserFetcher browserFetcher) =>
        await Puppeteer.LaunchAsync(new LaunchOptions
        {
            Headless = true,
            ExecutablePath = browserFetcher.GetExecutablePath(Chrome.DefaultBuildId)
        });
}
```
